service/controller/resource/statuscondition/create.go

```diff
@@ -45,23 +40,32 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		r.logger.LogCtx(ctx, "level", "debug", "message", "found latest cluster")
 	}
 
+	tenantClient, err := r.tenantClient.K8sClient(ctx, cr)
+	if tenantclient.IsNotAvailable(err) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "tenant client is not available yet")
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling reconciliation")
+		reconciliationcanceledcontext.SetCanceled(ctx)
```

Comment:
Would it be good enough to let each handler decide on its own? This now tells all the handlers they cannot proceed because this handler got a problem. There might be a handler that would be happy to do some work.

Reply:
So you mean not cancelling and return only?

Reply:
That would probably be the better alternative here. Only cancelling this handler and let the others decide what to do. When other handlers rely on the client they have to handle the unavailability themselves as well.